ark/attest/assert/chainableAssertions.ts

```diff
@@ -200,14 +202,53 @@ export class ChainableAssertions implements AssertionRecord {
 	}
 
 	get type(): any {
+		const declarationPrefix = "type T = "
```

Comment:
I would define these at the root of the module instead of in this getter's body so they don't get created each time `type` is accessed.

Reply:
Happy to, though I could use clarification -

I'm seeing a lot of cases here where functions are defined within class methods. Is the convention here to define non-function vars at the root level but relevant functions within the method where they're needed?

Or are you recommending I pull both the var and function definitions out of the `type` method?

Reply:
As a rule of thumb, any function that doesn't depend on the current class instance (i.e. this) should be moved outside the class altogether.

The distinction between whether the function is defined using the function keyword or as a variable with const isn't relevant to this case. The primary reasons are:

Avoid the extra overhead of creating the function on each function invocation (though this may be partially optimized away internally)
Decouple pure, testable logic (e.g. the formatting calls that just accept a string) from stateful methods that only exist within a `ChainableAssertions` instance.